src/vs/editor/contrib/anchorSelect/anchorSelect.ts

```diff
@@ -0,0 +1,172 @@
+import { IEditorContribution } from 'vs/editor/common/editorCommon';
+import { TrackedRangeStickiness } from 'vs/editor/common/model';
+import { MarkdownString } from 'vs/base/common/htmlContent';
+
+export const SelectionAnchorSet = new RawContextKey('selectionAnchorSet', false);
+
+class SelectionAnchorController implements IEditorContribution {
+
+	public static readonly ID = 'editor.contrib.selectionAnchorController';
+
+	static get(editor: ICodeEditor): SelectionAnchorController {
+		return editor.getContribution<SelectionAnchorController>(SelectionAnchorController.ID);
+	}
+
+	private decorationId: string | undefined;
+	private selectionAnchorSetContextKey: IContextKey<boolean>;
+
+	constructor(
+		private editor: ICodeEditor,
+		@IContextKeyService contextKeyService: IContextKeyService
+	) {
+		this.selectionAnchorSetContextKey = SelectionAnchorSet.bindTo(contextKeyService);
+	}
+
+	setSelectionAnchor(): void {
+		if (this.editor.hasModel()) {
+			const position = this.editor.getPosition();
+			const previousDecorations = this.decorationId ? [this.decorationId] : [];
+			const newDecorationId = this.editor.deltaDecorations(previousDecorations, [{
+				range: Selection.fromPositions(position, position),
+				options: {
+					stickiness: TrackedRangeStickiness.NeverGrowsWhenTypingAtEdges,
+					hoverMessage: new MarkdownString().appendText(localize('selectionAnchor', "Selection Anchor")),
+					className: 'selection-anchor'
+				}
+			}]);
+			this.decorationId = newDecorationId.length === 1 ? newDecorationId[0] : undefined;
+			this.selectionAnchorSetContextKey.set(!!this.decorationId);
+		}
+	}
+
+	goToSelectionAnchor(): void {
+		if (this.editor.hasModel() && this.decorationId) {
+			const anchorPosition = this.editor.getModel().getDecorationRange(this.decorationId);
+			if (anchorPosition) {
+				this.editor.setPosition(anchorPosition.getStartPosition());
+			}
+		}
+	}
+
+	selectFromAnchorToCursor(): void {
+		if (this.editor.hasModel() && this.decorationId) {
+			const start = this.editor.getModel().getDecorationRange(this.decorationId);
+			if (start) {
+				const end = this.editor.getPosition();
+				this.editor.setSelection(Selection.fromPositions(start.getStartPosition(), end));
+				this.cancelSelectionAnchor();
+			}
+		}
+	}
+
+	cancelSelectionAnchor(): void {
+		if (this.decorationId) {
+			this.editor.deltaDecorations([this.decorationId], []);
+			this.decorationId = undefined;
+			this.selectionAnchorSetContextKey.set(false);
+		}
+	}
+
+	dispose(): void {
+		this.cancelSelectionAnchor();
+	}
+}
+
+class SetSelectionAnchor extends EditorAction {
+	constructor() {
+		super({
+			id: 'editor.action.setSelectionAnchor',
+			label: localize('setSelectionAnchor', "Set Selection Anchor"),
+			alias: 'Set Selection Anchor',
+			precondition: undefined,
+			kbOpts: {
+				kbExpr: EditorContextKeys.editorTextFocus,
+				primary: KeyChord(KeyMod.CtrlCmd | KeyCode.KEY_K, KeyCode.KEY_B),
```

Comment:
We should look if `cmd+k cmd+b` is available. You can check the availability across OSes here -- https://github.com/microsoft/vscode-docs/tree/master/build/keybindings